coalib/output/printers/LogPrinter.py

```diff
@@ -1,54 +1,194 @@
+            self.printer.print(msg)
+        else:
+            sys.stdout.write(msg + "\n")
+
+
+class ColoredFormatter(logging.Formatter):
+    """
+    Formatter for logging which adds colors to the output by inserting special
+    symbols where appropriate.
+    """
+
+    def __init__(self, *args, **kwargs):
+        """
+        :param use_color:  If True highlights records levelname with respective
+                           color.
+        """
+        self.use_color = kwargs.pop('use_color', False)
+        super(ColoredFormatter, self).__init__(*args, **kwargs)
+
+    def format(self, record, *args, **kwargs):
+        if record.levelno in LOG_LEVEL_COLORS and self.use_color:
+            color_begin = LOG_LEVEL_COLORS[record.levelno]
+            color_end = Style.RESET_ALL
+        else:
+            color_begin = color_end = ''
+        record.levelname = '{cb}[{levelname}]{ce}'.format(
+            cb=color_begin,
+            ce=color_end,
+            levelname=record.levelname)
+        return super(ColoredFormatter, self).format(record, *args, **kwargs)
+
+
 class LogPrinter:
     """
     The LogPrinter class allows to print log messages to an underlying Printer.
 
     This class is an adapter, means you can create a LogPrinter from every
     existing Printer instance.
     """
+    _loggers = {}
+
+    @staticmethod
+    def _get_logger(printer, logger_name, time_fmt=None):
+        """
+        Returns existing or newly created and configured logger.
+        :param logger_name: name of the logger to be returned.
+        :param printer:     printer to initialize logger with if it doesn't
+                            exist, yet.
+        :param time_fmt:    format to be used for time output. No time is
+                            pinter if None.
+        """
+        if not LogPrinter._loggers.get(logger_name):
+            if time_fmt:
+                format_str = '%(levelname)s[%(asctime)s] %(message)s'
+            else:
+                format_str = '%(levelname)s %(message)s'
+
+            use_color = isinstance(printer, ColorPrinter)
+            formatter = ColoredFormatter(format_str,
+                                         time_fmt,
+                                         use_color=use_color)
+
+            handler = PrinterHandler(printer)
+            handler.setFormatter(formatter)
+
+            logger = logging.getLogger(logger_name)
+            logger.addHandler(handler)
+
+            LogPrinter._loggers[logger_name] = logger
+        return LogPrinter._loggers[logger_name]
 
     def __init__(self,
-                 printer,
+                 printer=None,
                  log_level=LOG_LEVEL.INFO,
-                 timestamp_format="%X"):
+                 timestamp_format="%X",
+                 of_type=True):
         """
-        Creates a new log printer from an existing Printer.
+        Creates a new log printer from an existing Printer or returns existing
+        one if it already exist.
```

Comment:
It may be a little early for the note, cause there is no alternative configured printing solution, yet. Not sure, if it's better to add in this PR configuring logging from files for different kinds of printers.